controller/job/backresthandler.go

```diff
@@ -0,0 +1,229 @@
+	// LABEL_BACKREST_RESTORE label to determine if the restore is part of and ongoing clone.
+	switch {
+	case labels[config.LABEL_BACKREST_COMMAND] == "backup":
+		c.handleBackrestBackupUpdate(job)
+	case labels[config.LABEL_PGO_CLONE_STEP_2] == "true":
+		c.handleCloneBackrestRestoreUpdate(job)
+	case labels[config.LABEL_BACKREST_RESTORE] == "true":
+		c.handleBackrestRestoreUpdate(job)
+	case labels[config.LABEL_BACKREST_COMMAND] == crv1.PgtaskBackrestStanzaCreate:
+		c.handleBackrestStanzaCreateUpdate(job)
+	}
+
+	return nil
+}
+
+// handleBackrestRestoreUpdate is responsible for handling updates to backrest restore,
+// (specifically those that have not been submitted as part of the clone process)
+func (c *Controller) handleBackrestRestoreUpdate(job *apiv1.Job) error {
+
+	labels := job.GetObjectMeta().GetLabels()
+	log.Debugf("jobController onUpdate backrest restore job case")
+	log.Debugf("got a backrest restore job status=%d", job.Status.Succeeded)
+	log.Debugf("set status to restore job completed  for %s", labels[config.LABEL_PG_CLUSTER])
+	log.Debugf("workflow to update is %s", labels[crv1.PgtaskWorkflowID])
+	if err := util.Patch(c.JobClient, patchURL, crv1.JobCompletedStatus, patchResource, job.Name,
+		job.ObjectMeta.Namespace); err != nil {
+		log.Error("error in patching pgtask " + labels[config.LABEL_JOB_NAME] + err.Error())
+	}
+
+	backrestoperator.UpdateRestoreWorkflow(c.JobClient, c.JobClientset, labels[config.LABEL_PG_CLUSTER],
+		crv1.PgtaskWorkflowBackrestRestorePVCCreatedStatus, job.ObjectMeta.Namespace, labels[crv1.PgtaskWorkflowID],
+		labels[config.LABEL_BACKREST_RESTORE_TO_PVC], job.Spec.Template.Spec.Affinity)
+	publishRestoreComplete(labels[config.LABEL_PG_CLUSTER], job.ObjectMeta.Labels[config.LABEL_PG_CLUSTER_IDENTIFIER], job.ObjectMeta.Labels[config.LABEL_PGOUSER], job.ObjectMeta.Namespace)
+
+	return nil
+}
+
+// handleBackrestRestoreUpdate is responsible for handling updates to backrest restore jobs that
+// have been submitted in order to clone a cluster
+func (c *Controller) handleCloneBackrestRestoreUpdate(job *apiv1.Job) error {
+
+	log.Debugf("jobController onUpdate clone step 2 job case")
+	log.Debugf("clone step 2 job status=%d", job.Status.Succeeded)
+
+	if job.Status.Succeeded == 1 {
+		namespace := job.ObjectMeta.Namespace
+		sourceClusterName := job.ObjectMeta.Annotations[config.ANNOTATION_CLONE_SOURCE_CLUSTER_NAME]
+		targetClusterName := job.ObjectMeta.Annotations[config.ANNOTATION_CLONE_TARGET_CLUSTER_NAME]
+		workflowID := job.ObjectMeta.Labels[config.LABEL_WORKFLOW_ID]
+
+		log.Debugf("workflow to update is %s", workflowID)
+
+		// first, make sure the Pgtask resource knows that the job is complete,
+		// which is using this legacy bit of code
+		if err := util.Patch(c.JobClient, patchURL, crv1.JobCompletedStatus, patchResource, job.Name, namespace); err != nil {
+			log.Error(err)
+			// we can continue on, even if this fails...
```

Comment:
I'd recommend a `Warn` then if we can continue on.

Reply:
Which I realize I may be commenting on my own code 😆

Reply:
Changed to `Warn`.